docs/create-analystics-event.md

```diff
@@ -0,0 +1,143 @@
+
+To implement an analytics event in the extension, you'll need to use the `MetaMetricsContext` and its `trackEvent` function. This context provides a standardized way to track events while handling opt-in status and data anonymization automatically. Below is an example showing how to implement event tracking in a React component:
+
+```typescript
+// 1. Import the required types and context
+import { MetaMetricsContext } from '../../../shared/contexts/metametrics';
+import {
+  MetaMetricsEventName,
+  MetaMetricsEventCategory,
+  MetaMetricsEventFragment,
+} from '../../../shared/constants/metametrics';
+
+// 2. Use the context in your component
+function YourComponent() {
+  const { trackEvent } = useContext(MetaMetricsContext);
+
+  const handleAction = () => {
+    trackEvent({
+      event: MetaMetricsEventName.YourNewEventName,
+      category: MetaMetricsEventCategory.YourCategory,
+      properties: {
+        // Add properties that provide context about the event
+        property_one: 'value_one',
+        property_two: 'value_two',
+      },
+      sensitiveProperties: {
+        // Add properties that might contain sensitive data
+        // These will be automatically anonymized
+        wallet_address: '0x...',
+      },
+    });
+  };
+
+  return <button onClick={handleAction}>Perform Action</button>;
+}
+```
+
+#### 2.3 E2E Testing
+
+For implementing analytics event mocks in E2E tests using `testSpecificMock`, please refer to recording in notion of [Extension Codebase - E2E tests (mocking)](https://www.notion.so/metamask-consensys/f649ecf027cc41db858d3a3574fe3a99?v=da4af5e48f0e44d7b4536c4e2e911f36).
+
+Example E2E test implementation:
+
+```typescript
+// Mock the analytics endpoint
+async function mockAnalyticsEndpoint(mockServer) {
+  return await mockServer
+    .forPost('https://api.segment.io/v1/batch')
+    .withJsonBodyIncluding({
+      batch: [
+        {
+          type: 'track',
+          event: MetaMetricsEventName.YourNewEventName,
+          properties: {
+            property_one: 'value_one',
+            property_two: 'value_two',
+          },
+        },
+      ],
+    })
+    .thenCallback(() => ({ statusCode: 200 }));
+}
+
+describe('YourNewEventName Analytics', () => {
+  it('should track the event with correct properties', async () => {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder()
+          .withMetaMetricsController({
+            participateInMetaMetrics: true,
+          })
+          .build(),
+        testSpecificMock: async (mockServer) => [
+          await mockAnalyticsEndpoint(mockServer),
+        ],
+      },
+      async ({ driver, mockedEndpoints }) => {
+        // Test implementation
+        const events = await getEventPayloads(driver, mockedEndpoints);
+        assert.equal(events[0].properties.property_one, 'value_one');
+        assert.equal(events[0].properties.property_two, 'value_two');
+      },
+    );
+  });
+});
+```
+
+#### 2.4. Validate the Event
```

Comment:
S: optional if user has access, but helps sometimes to detect issues, we should also check for the event in a Mixpanel dev dashboard.

Reply:
They have to access it (only for internal) using their Okta dashboard.

Reply:
Oh my bad, i mean link of Mixpanel document ...